lib/kioskPrinter.js

```diff
@@ -0,0 +1,241 @@
+const SerialPort = require('serialport')
+
+module.exports = {
+    checkStatus,
+    printWallet
+}
+
+
+// Public Interface
+// ----------------
+
+function checkStatus (printerCfg, statusQueryTimeout) {
+  return new Promise((resolve, reject) => {
+    const model = checkPrinterModel(printerCfg)
+
+    if (model === printerModels.zebra)
+      return checkStatusZebra(printerCfg, statusQueryTimeout)
+        .then((printerStatus) => resolve(printerStatus))
+        .catch((err) => reject(err))
+
+    return reject(errorMsgUnsupportedPrinter())
+  })
+}
+
+
+function printWallet (wallet, printerCfg) {
+  return new Promise((resolve, reject) => {
```

Comment:
We don't need to create our own Promise here because we're going to be calling functions that already return promises (e.g., ``printWalletZebra()``).